nucypher/blockchain/eth/actors.py

```diff
@@ -692,6 +601,94 @@ def encrypt_threshold_decryption_response(
+        return self.__staking_provider_address
+
+    def get_staking_provider_address(self):
+        self.__staking_provider_address = (
+            self.application_agent.get_staking_provider_from_operator(
+                self.operator_address
+            )
+        )
+        self.checksum_address = self.__staking_provider_address
+        self.nickname = Nickname.from_seed(self.checksum_address)
+        return self.__staking_provider_address
+
+    @property
+    def is_confirmed(self):
+        return self.application_agent.is_operator_confirmed(self.operator_address)
+
+    def block_until_ready(self, poll_rate: int = None, timeout: int = None):
+        emitter = StdoutEmitter()
+        client = self.application_agent.blockchain.client
+        poll_rate = poll_rate or self.READY_POLL_RATE
+        timeout = timeout or self.READY_TIMEOUT
+        start, funded, bonded = maya.now(), False, False
+        while not (funded and bonded):
+            if timeout and ((maya.now() - start).total_seconds() > timeout):
+                message = f"x Operator was not qualified after {timeout} seconds"
+                emitter.message(message, color="red")
+                raise self.ActorError(message)
+
+            if not funded:
+                # check for funds
+                ether_balance = client.get_balance(self.operator_address)
+                if ether_balance:
+                    # funds found
+                    funded, balance = True, Web3.from_wei(ether_balance, "ether")
+                    emitter.message(
+                        f"✓ Operator {self.operator_address} is funded with {balance} ETH",
+                        color="green",
+                    )
+                else:
+                    emitter.message(
+                        f"! Operator {self.operator_address} is not funded with ETH",
+                        color="yellow",
+                    )
+
+            if (not bonded) and (self.get_staking_provider_address() != NULL_ADDRESS):
+                bonded = True
+                emitter.message(
+                    f"✓ Operator {self.operator_address} is bonded to staking provider {self.staking_provider_address}",
+                    color="green",
+                )
+            else:
+                emitter.message(
+                    f"! Operator {self.operator_address } is not bonded to a staking provider",
+                    color="yellow",
+                )
+
+            time.sleep(poll_rate)
+
+        if not self.is_confirmed:
+            emitter.message(
+                f"! Publishing provider public key for {self.staking_provider_address}",
+                color="yellow",
+            )
+            receipt = self.set_provider_public_key()
+            txhash = receipt["transactionHash"].hex()
+            emitter.message(
+                f"✓ Successfully published provider public key for {self.staking_provider_address} (TX: {txhash})",
```

Comment:
It would be useful to explain that this happens on Polygon Mainnet/Mumbai

Reply:
✔️